### ISSUE TITLE
SCM - Differentiate between submodules and normal repos

### ISSUE THREAD

**Issue by firelizzard18** [microsoft/vscode#43951]
### Issue Type
Feature Request

**I would like submodules to be nested underneath their parent repo in the Source Control Pane.**

### Description

I love the new submodule feature, but there are a couple of issues with the Source Control pane:

  * I can't tell which providers (repos) are submodules
  * I can't tell what top-level provider a given submodule provider is associated with

For most situations, this probably isn't an issue. But for some of my projects, it is a serious issue. For example, I have projects:

  * A - depends on nothing
  * B - depends on A
  * C - depends on A and B
  * etc

For development, I use a multi-root workspace with a directory structure something like this:

  * Projects
    * A
    * B *(references `../A`)*
    * C *(references `../A` and `../B`)*

For continuous integration and deployment, I use submodules:

  * C *(references `./.libs/A` and `./.libs/B`)*
    * .libs
      * A
      * B *(references `/builds/C/.libs/A`)*

While I could leave the submodules uninitialized on my dev machine, my workflow includes doing a build with submodules before I push, to ensure I haven't forgotten to update a submodule, because why waste CPU cycles on a CI build that's guaranteed to fail.

In this particular example, where B has A as a submodule, and C has A and B, I will have six repos (A, B, C, B/.libs/A, C/.libs/A, and C/.libs/B) show up in VSCode. While I can and do disable submodule discovery for these projects, that doesn't stop VSCode from picking them up when I open a file in a submodule.

### VS Code Info

VS Code version: Code 1.20.1 (f88bbf9137d24d36d968ea6b2911786bfe103002, 2018-02-13T15:34:36.336Z)
OS version: Windows_NT x64 10.0.16299

**Comment by solomon87** [comment]
Plus one for me. When I work in the parent directory of all my git projects, it causes all of my submodules to display as many times as they are included within other projects. An option to not display submodules in the source control providers list would be helpful.

**Comment by michael-letcher** [comment]
I have the same issue, which is worsened when using workspaces.

![image](https://user-images.githubusercontent.com/4531036/55444457-42abf600-5602-11e9-97a3-c0aba708631e.png)

**Comment by joaomoreno** [comment]
#40548

**Comment by NotWearingPants** [comment]
@joaomoreno This is **not** a duplicate of #40548

You are the one who changed the title in August from `Differentiate between submodules and normal repos` to `Differentiate between repositories with the same name`, which is not the same.

Please **reopen**, this is still an issue

**Comment by firelizzard18** [comment]
@joaomoreno I totally agree with @NotWearingPants. My issue is specifically related to submodules. #40548 would lessen the issue, but would not solve it.

**Comment by joaomoreno** [comment]
You're absolutely right, sorry about that.

**Comment by Awbmilne** [comment]
Duplicate of #101859 : PLEASE UPVOTE IT THERE SO IT CAN BE WORKED ON!

**Comment by wolfsilver** [comment]
This extension can help solve this problem.

https://marketplace.visualstudio.com/items?itemName=R.git-submodule-sort

**Comment by jasonbeach** [comment]
In this same vein, I've started to use multiroot workspaces pretty extensively (they're awesome....I no longer need to have several vscode windows open).  The projects I'm working on span a number of git repos and so I have a few top level folders that I've added to my workspace and underneath those I have all my repos placed. The Explorer view has a nice tree structure so this works out nice and makes it very easy to jump between repos.  However, the source control tab doesn't show that structure--it flattens all the repos out (even repos that are submodules). Is there anyway to get the source control pane to follow the folder structure the repos are organized into?

**Comment by mjbogusz** [comment]
+1 on this. When working on a ROS workspace it's common to have multiple repos potentially with repeating submodules and the current view doesn't allow easy differentiating between them - see attached screenshot. The only way I've found is to hover the mouse over a repo's name, which displays it's full path.

<details>
  <summary>Screenshot</summary>

  ![image](https://user-images.githubusercontent.com/4386503/199466014-a78d6155-5d2b-4157-be95-aa853d322e0b.png)

</details>

**Comment by p-servus** [comment]
i also needs the fix for that .. if i have two subModules in this directories:
- modules/vendorA/utils
- modules/vendorB/utils
.. then it will be displayed like this:
- utils
- utils

.. i can sort them by path (in my user settings.json):
	"scm.repositories.sortOrder": "path",
.. but there is no option do display as path!

**Comment by starball5** [comment]
Note: Related on Stack Overflow: [How can I show Git submodules indented and underneath their parent repository in VS Code](https://stackoverflow.com/q/76869937/11107541)

**Comment by Knar33** [comment]
I have a vscode workspace that has the submodule repo plus two other repos that use the submodule, so the submodule repo shows up 3 times in my repositories list! Would be nice to have an option to completely hide submodule repos at least.

**Comment by robertoandrade** [comment]
this is the biggest pet peeve I have with using vscode for scm! since they sort of have resolved the 3-way merge view relatively recently.

**Comment by lszomoru** [comment]
Thank you all for your patience regarding thie feature request. The latest VS Code Insiders release contains changes that shows the opened source control providers (repositories, submodules, and worktrees) in a hierancy along with distinct icons. We would really appreciate if you would download the latest VS Code Insiders release and let us know what you think about the changes. Thanks!

<img width="616" height="211" alt="Image" src="https://github.com/user-attachments/assets/bd9d9384-b5a7-435c-aa45-32035840be19" />

**Comment by lszomoru** [comment]
Verification steps:
* Install latest VS Code Insiders release
* Ensure that the `git.detectSubmodules` setting is enabled
* Open a folder/workspace that contains a git repository with a submodule
* Open the Repositories view
    * Confirm that both the repository and the submodule are listed
    * Confirm that the submodule appears as a child of the repository
    * Confirm that the repository/submodule have different icons

**Comment by JonathanDevelops** [comment]
Sadly this is only working partly for me in the Insiders build. 

<img width="77" height="182" alt="Image" src="https://github.com/user-attachments/assets/2bd39866-a1b0-477f-84da-021ceaf3c2c3" />

I do not see the hirachy
I see diferent labels

hirachy would be like this
-
--
---
--

I used version 1.103.0-insider

Edit: correction on my side: this problem occures only on my virtual drive, created over:

NET share "Z:"=<path>
SUBSET "Z:" .\..

**Comment by lszomoru** [comment]
@JonathanDevelops, could you please file a separate issues for that with a detailed description of steps to reproduce the issue? Could you also share the contents of the `.gitmodules` file? I am interested in the path format being used there. Thanks!
